gapis/trace/tracer/tracer.go

```diff
@@ -70,6 +71,20 @@ type Tracer interface {
 	GetDevice() bind.Device
 }
 
+// Parse the perfetto options, and returns the required layers
+func LayersFromOptions(ctx context.Context, o *service.TraceOptions) []string {
+	ret := []string{}
+	if o.PerfettoConfig == nil {
+		return ret
+	}
+	for _, x := range o.PerfettoConfig.GetDataSources() {
+		if _, err := layout.LibraryFromLayerName(x.GetConfig().GetName()); err == nil {
+			ret = append(ret, x.GetConfig().GetName())
+		}
+	}
+	return ret
+}
+
```

Comment:
Yikes. I don't like this. It does expose that we have a bit of an issue with where knowledge exists about trace config vs env setup. For now, to keep with established pattern, let's have the UI create the config proto, but then add the list of layers to add as an option in the trace request. What do you think? We can then possibly find a better way to do this in the future, I'm even open to moving the entire Perfetto config construction into the server...

Reply:
Actually, after talking F2F, this is OK and we can keep it and key of the layers to enable based on the enabled data sources. This simplifies, for example, taking a trace via `gapit`, as it won't need to be updated every time a new layer is added.